android/kotlin/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/IngestionMetadata.kt

```diff
@@ -0,0 +1,29 @@
+package com.amplitude.ampli
+
+import com.amplitude.common.jvm.ConsoleLogger
+import org.json.JSONException
+import org.json.JSONObject
+
+class IngestionMetadata (sourceName: String, sourceVersion: String) {
+     var sourceName : String
+     var sourceVersion : String
+    init {
+        this.sourceName = sourceName
+        this.sourceVersion = sourceVersion
+    }
+
+    fun toJsonObject(): JSONObject {
+        var jsonObject = JSONObject()
+        try {
+            if (!sourceName.isNullOrEmpty()) {
```

Comment:
Is this necessary with Koltins null safety? https://kotlinlang.org/docs/null-safety.html#nullable-types-and-non-null-types